Crash with newest update

**Describe the bug**
The app crashes immediately after the last PR merge and update from 2022-05-25. Uninstalling and installing the app shows the login screen, but after the login we got crashes again.

**To Reproduce**
Steps to reproduce the first behavior:
1. Update to the newest version of the app (app must be already present on your phone)
2. Open it
3. See crash

Steps to reproduce the second behavior:
1. Uninstall and install the newest version of the app
2. Open it
3. Login
4. See crash

NB: Only the second behavior is visible in crashlytics, I guess the first crashes too early.


Got this traceback on my phone:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.android.feedback/com.google.android.feedback.FeedbackActivity}: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3761)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3935)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2286)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8292)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
Caused by: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.os.Parcel.readException(Parcel.java:2334)
	at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkTypeForSubscriber(ITelephony.java:9300)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2971)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2935)
	at com.google.android.feedback.FeedbackSession.getTelephonyInfo(FeedbackSession.java:307)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:152)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:141)
	at com.google.android.feedback.FeedbackActivity.pushSession(FeedbackActivity.java:209)
	at com.google.android.feedback.FeedbackActivity.createFeedbackSession(FeedbackActivity.java:178)
	at com.google.android.feedback.FeedbackActivity.onCreate(FeedbackActivity.java:96)
	at android.app.Activity.performCreate(Activity.java:8241)
	at android.app.Activity.performCreate(Activity.java:8207)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3734)
	... 12 more

```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -33,8 +33,8 @@ jobs:
       - name: Java setup
         uses: actions/setup-java@v3
         with:
-          distribution: 'zulu' # See 'Supported distributions' for available options
-          java-version: '11'
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '17'
       - name: Mock google-services.json
         run: mv app/google-services-mock.json app/google-services.json
       - name: Instrumentation Tests

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -38,7 +38,7 @@ jobs:
       - name: Mock google-services.json
         run: mv app/google-services-mock.json app/google-services.json
       - name: Instrumentation Tests
-        uses: reactivecircus/android-emulator-runner@v3
+        uses: reactivecircus/android-emulator-runner@v2
         with:
           api-level: 29
           # If you want to test only a specific variant, use one of the following:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -34,7 +34,7 @@ jobs:
         uses: actions/setup-java@v3
         with:
           distribution: 'temurin' # See 'Supported distributions' for available options
-          java-version: '17'
+          java-version: '11'
       - name: Mock google-services.json
         run: mv app/google-services-mock.json app/google-services.json
       - name: Instrumentation Tests

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Mock google-services.json
         run: mv app/google-services-mock.json app/google-services.json
       - name: Test the app
@@ -29,16 +29,16 @@ jobs:
     runs-on: macos-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Java setup
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           distribution: 'zulu' # See 'Supported distributions' for available options
           java-version: '11'
       - name: Mock google-services.json
         run: mv app/google-services-mock.json app/google-services.json
       - name: Instrumentation Tests
-        uses: reactivecircus/android-emulator-runner@v2
+        uses: reactivecircus/android-emulator-runner@v3
         with:
           api-level: 29
           # If you want to test only a specific variant, use one of the following:
@@ -53,7 +53,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Inject google-services.json
         env:
           GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}

--- a/app/src/main/java/it/bz/noi/community/data/models/News.kt
+++ b/app/src/main/java/it/bz/noi/community/data/models/News.kt
@@ -1,11 +1,13 @@
 package it.bz.noi.community.data.models
 
 import android.os.Parcelable
+import androidx.annotation.Keep
 import com.google.gson.annotations.SerializedName
 import it.bz.noi.community.utils.Utils
 import kotlinx.parcelize.Parcelize
 import java.util.*
 
+@Keep
 @Parcelize
 data class News(
 	@SerializedName("Id")
@@ -17,43 +19,47 @@ data class News(
 	@SerializedName("ContactInfos")
 	val contactInfo: Map<String, ContactInfo>,
 	@SerializedName("ImageGallery")
-	val images: List<NewsImage>?,
+	val images: List<NewsImage>? = null,
 	@SerializedName("ODHTags")
-	val tags: List<Tag>?
+	val tags: List<Tag>? = null
 ) : Parcelable
 
+@Keep
 @Parcelize
 data class Tag(
 	@SerializedName("Id")
-	val id: String?
+	val id: String? = null,
 ) : Parcelable
 
+@Keep
 @Parcelize
 data class Detail(
 	@SerializedName("Title")
-	val title: String?,
+	val title: String? = null,
 	@SerializedName("AdditionalText")
-	val abstract: String?,
+	val abstract: String? = null,
 	@SerializedName("BaseText")
-	val text: String?,
+	val text: String? = null,
 ) : Parcelable
 
+@Keep
 @Parcelize
 data class ContactInfo(
 	@SerializedName("CompanyName")
-	val publisher: String?,
+	val publisher: String? = null,
 	@SerializedName("LogoUrl")
-	val logo: String?,
+	val logo: String? = null,
 	@SerializedName("Url")
-	val externalLink: String?,
+	val externalLink: String? = null,
 	@SerializedName("Email")
-	val email: String?,
+	val email: String? = null,
 ) : Parcelable
 
+@Keep
 @Parcelize
 data class NewsImage(
 	@SerializedName("ImageUrl")
-	val url: String?
+	val url: String? = null
 ) : Parcelable
 
 fun News.getDetail(): Detail? {

